GSA-TTS/FAC
# More Advanced Search Parameters & HTML Refactor

# More Advanced Search Parameters & HTML Refactor

Issues: https://github.com/GSA-TTS/FAC/issues/3576, https://github.com/GSA-TTS/FAC/issues/3622

## Changes:
1. Keep search in one HTML file (`search.html`, instead of `search.html` + `advanced.html`). Give every filter its own file, and conditionally render them depending on which page you're on.
2. Add a few new filters, and change one:
    * Fiscal year end month (basic + advanced)
    * Type requirement (advanced)
    * Entity type (basic + advanced)
    * Passthrough name (advanced)
    * Report ID (basic + advanced)
    * UEI/EIN field searches the additional UEI/EIN tables in addition to general
3. Unit tests for the new filters

## How to test:
1. Make sure you have some data loaded locally, and that your materialized view is up to date. 
2. Switch to this branch and navigate to search.
3. Try the the new filters, and verify accuracy:
    * Fiscal year end month. Searches on currently selected audit years. Most audits are either June or December. You can check a years results and verify that about half come back when using these months. 
    * Type requirement. The five most common are "P", "N", "L", "I", and "B". But, there are a huge number of valid options, and they can chain together. A good one is "ABCFGHIJLMN". 
    * Entity type. It doesn't seem like there are a lot of controls on this field, because "state" comes back with a lot of audits. But, it's what's the data shows. 
    * Passthrough name. Find a passthrough name locally (like "SDE"), and verify that the relevant report_ids come back in a search on it. Sometimes, there are a lot of results on one passthrough name.
    * Report ID. Try a bunch!
    * New UEI/EIN behavior. Find an audit with additional UEIs/EINs, and search on them to verify that the record now comes in, even if its general object does not represent these values. 

## Screenshots:
The new filters:
![image](https://github.com/GSA-TTS/FAC/assets/91098850/907f8ec4-1a97-473c-ac7e-b8e9d778c7b5)

Most common type requirements in Census public data (~550 different combinations are represented):
<img width="347" alt="image" src="https://github.com/GSA-TTS/FAC/assets/91098850/0cfbc33a-6550-400d-9a2c-d0acebc46ead">

## PR checklist: submitters


## PR checklist: reviewers


The larger the PR, the stricter we should be about these points.
